code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+	var/tritium = 0
+	var/hydrogen = 0
+	var/helium = 0
+
+	var/m_plasma = 0
+	var/m_nitrogen = 0
+	var/m_oxygen = 0
+	var/m_co2 = 0
+	var/m_h2o = 0
+	var/m_n2o = 0
+	var/m_no2 = 0
+	var/m_freon = 0
+	var/m_bz = 0
+	var/m_proto_nitrate = 0
+	var/m_healium = 0
+	var/m_zauker = 0
+	var/m_antinoblium = 0
+	var/m_hypernoblium = 0
+
+	///Check if the user want to remove the waste gases
+	var/waste_remove = FALSE
+	///User controlled variable to control the flow of the fusion by changing the contact of the material
+	var/heating_conductor = 100
+	///User controlled variable to control the flow of the fusion by changing the volume of the gasmix by controlling the power of the magnetic fields
+	var/magnetic_constrictor  = 100
+	///User controlled variable to control the flow of the fusion by changing the instability of the reaction
+	var/current_damper = 0
+	///Stores the current fusion mix power level
+	var/power_level = 0
+	///Stores the iron content produced by the fusion
+	var/iron_content = 0
+	///User controlled variable to control the flow of the fusion by changing the amount of fuel injected
+	var/fuel_injection_rate = 250
+	///User controlled variable to control the flow of the fusion by changing the amount of moderators injected
+	var/moderator_injection_rate = 250
+	///Used to make the hypertorus fusion process tick once per second
+	COOLDOWN_DECLARE(hypertorus_reactor)
+
+	///Integrity of the machine, if reaches 900 the machine will explode
+	var/critical_threshold_proximity = 0
+	///Store the integrity for calculations
+	var/critical_threshold_proximity_archived = 0
+	///Our "Shit is no longer fucked" message. We send it when critical_threshold_proximity is less then critical_threshold_proximity_archived
+	var/safe_alert = "Main containment field returning to safe operating parameters."
+	///The point at which we should start sending messeges about the critical_threshold_proximity to the engi channels.
+	var/warning_point = 50
+	///The alert we send when we've reached warning_point
+	var/warning_alert = "Danger! Magnetic containment field faltering!"
+	///The point at which we start sending messages to the common channel
+	var/emergency_point = 700
+	///The alert we send when we've reached emergency_point
+	var/emergency_alert = "HYPERTORUS MELTDOWN IMMINENT."
+	///The point at which we melt
+	var/melting_point = 900
+	///Boolean used for logging if we've passed the emergency point
+	var/has_reached_emergency = FALSE
+	///Time in 1/10th of seconds since the last sent warning
+	var/lastwarning = 0
+
+	///Our internal radio
+	var/obj/item/radio/radio
+	///The key our internal radio uses
+	var/radio_key = /obj/item/encryptionkey/headset_eng
+	///The engineering channel
+	var/engineering_channel = "Engineering"
+	///The common channel
+	var/common_channel = null
+
+	///Our soundloop
+	var/datum/looping_sound/hypertorus/soundloop
+	///cooldown tracker for accent sounds
+	var/last_accent_sound = 0
```

Comment:
I'm a little worried about copying too much from the sm, tho I get the intent.

Reply:
there are only a few places that are copied from the SM, but those are the best way to deal with that particular thing